lib/index.js

```diff
@@ -370,14 +426,28 @@ exports.getStorageClient = (ctx) => {
     const url = `${ctx.configuration.fusebit_storage_audience}/v1/account/${ctx.configuration.fusebit_storage_account_id}/subscription/${ctx.configuration.fusebit_storage_subscription_id}/storage/${ctx.configuration.fusebit_storage_id}`;
 
     return {
+        etag: null,
         get: async () => {
             const response = await Superagent.get(url)
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
+            this.etag = response.headers.etag;
             return response.status === 404 ? undefined : response.body.data;
         },
-        put: async (data) => {
-            await Superagent.put(url).set('Authorization', `Bearer ${accessToken}`).send({ data });
+        put: async (data, oldEtag) => {
+            let request = Superagent.put(url).set('Authorization', `Bearer ${accessToken}`);
+            if (oldEtag){
+              request = request.set('If-Match', oldEtag);
```

Comment:
Do not send the etag in the header. Send etag as part of the request body: `{ etag: '...', data: ...}`.